### ISSUE TITLE
Build error

### ISSUE THREAD

**Issue by jielemon** [terjeio/ioSender#259]
A newbie in visual studio.

Downloaded the latest revision of the ioSender source and opened ioSender.sln with Visual Studio 2022. There are 65 errors when building.  Here is the screenshot of some of errors.
![image](https://user-images.githubusercontent.com/109292090/206961350-df1c5f29-47c0-4a5b-8006-77cad7a3e5f0.png).

Could you help with that?

Thanks a lot.

**Comment by Jzhenli** [comment]
@terjeio I'm newer to this project, and build it error too.  How to install the missing the dependency in the VS IDE?

**Comment by terjeio** [comment]
@Jzhenli See [the wiki](https://github.com/terjeio/ioSender/wiki/Building).

**Comment by Jzhenli** [comment]
@terjeio Hi, where to find these libs, I check the latest code folder and cannot find them.
![image](https://user-images.githubusercontent.com/42105688/216908088-8f634bb1-8dd7-49fb-bd29-050f6bce2e13.png)

**Comment by terjeio** [comment]
You have to download them from the provided links or extract them from the latest [release](https://github.com/terjeio/ioSender/releases).

**Comment by Jzhenli** [comment]
Do you mean to download the pre-build version from the release？ I know that will works well, but now I want to know how build them from the VS IDE by myself, and I have met the same error as the topic image shows.

**Comment by Jzhenli** [comment]
@terjeio Do you mean to download the pre-build version from the release？ I know that will works well, but now I want to know how build them from the VS IDE by myself, and I have met the same error as the topic image shows.

**Comment by terjeio** [comment]
> Do you mean to download the pre-build version from the release?

Yes, that is an alternative to get the libraries if not downloading via the links - copy the relevant files as described in the wiki, they can be found inside the release folder and, IMO, have pretty obvious names.

**Comment by terjeio** [comment]
In the root folder of the sender projects I have:

_CNC Controls Camera\bin\Release_ folder with
```
AForge.Video.DirectShow.xml
AForge.Video.dll
AForge.Video.xml
AForge.xml
AForge.dll
AForge.Video.DirectShow.dll
```
Referenced by _CNC Controls Camera_ project. This is a bit odd - cannot remember why now. It may be better to put these in a _aforge_ folder in the root folder of the project and reference the .dll files anew. Or disable the camera as described in the Wiki.

_helix-toolkit_ folder with
```
HelixToolkit.Wpf.dll
HelixToolkit.Wpf.XML
```
Referenced in the _CNC GCodeViewer_ and _CNC Controls Probing_ projects.

_Vector-master\RP.Math.Vector3\bin\Release_ folder with
`RP.Math.dll`
_Vector-master\RP.Math.Vector3\bin\Release\RP.Math_ with
`Vector3.dll`
Referenced in the _CNC Controls Probing_ project.

_websocket-sharp\websocket-sharp\bin\Release_ folder with:
`websocket-sharp.dll`
Referenced in the _CNC Core_ project.

Hope this helps.

**Comment by Jzhenli** [comment]
@terjeio Have tried as your method, still cannot build the code. Would you told how to add the Vector lib? Seems cannot find it in the NuGet. The other libs I have installed as bellow image shows.
![image](https://user-images.githubusercontent.com/42105688/217125890-145f263c-6cae-46f1-b2e4-c15b73744a26.png)

**Comment by terjeio** [comment]
@Jzhenli Deleting and readding the references to the vector libs failed? Thats odd. Download the source and compile as a separate project and then delete and readd the references to the release build .dlls.

**Comment by Jzhenli** [comment]
@terjeio the setps I do as follows,
1. download source code ioSender-2.0.41.zip
2. unzip code and open ioSender.sln with vs2019 (os win11)
3. install below libs with NuGet, the version as bellow image shows
![image](https://user-images.githubusercontent.com/42105688/217180349-b540dba0-d749-4cce-9eb1-448ec171cfeb.png)

4. build code and 2 Erros messages
![image](https://user-images.githubusercontent.com/42105688/217180736-c6725154-49fd-441c-8444-eb99e461545a.png)

I don't know the reason why it fails, would help to check the steps? I guess the lib version may have some issues.

Thanks.

**Comment by terjeio** [comment]
It seems I have left references to the XCeed toolkit in the published source - a missing commit on my side.
Replace the two files in the attached zip in the _\CNC GCodeViewer\CNC GCodeViewer_ folder and try again.

[CNC GCodeViewer.zip](https://github.com/terjeio/ioSender/files/10673136/CNC.GCodeViewer.zip)

**Comment by terjeio** [comment]
_ToolIndicator.cs_ is experimental code - exclude it from the project by right clicking on the file in the Solution Explorer and select _Exclude From Project_.

**Comment by Jzhenli** [comment]
@terjeio thanks for your reply, now it's OK.
